src/ray/raylet/node_manager.cc

```diff
@@ -477,7 +477,24 @@ void NodeManager::ProcessNewClient(LocalClientConnection &client) {
   client.ProcessMessages();
 }
 
-void NodeManager::DispatchTasks() {
+void NodeManager::DispatchTasks(const Task* new_ready_task) {
```

Comment:
Let's try to avoid using raw pointers.

How about we instead pass in a `const std::list<Task> &task_queue`. If there is only one newly ready task, we can pass in the list `{new_ready_task}`, otherwise it's `local_queues_.GetReadyTasks()`.

Reply:
ok.

Reply:
Hmm actually what I meant was to pass in `local_queues_.GetReadyTasks()` directly, like this: `DispatchTasks(local_queues_.GetReadyTasks())`